src/client.rs

```diff
@@ -146,21 +152,46 @@ fn main() {
             match event {
                 Event::Response((addr, opt_packet)) => {
                     let packet = opt_packet.unwrap();
-                    println!("Got packet from server {:?}: {:?}", addr, packet);
+                    println!("DEBUG: Got packet from server {:?}: {:?}", addr, packet);
                     match packet {
                         Packet::Response{sequence, request_ack, code} => {
                             // XXX sequence
                             // XXX request_ack
                             match code {
                                 ResponseCode::OK => {
-                                    println!("OK :)");
+                                    if let Some(ref last_action) = client_state.last_req_action {
+                                        match last_action {
+                                            &RequestAction::JoinGameSlot(ref slot_name) => {
+                                                client_state.in_game = true;
+                                                client_state.game_slot = Some(slot_name.clone());
+                                                println!("Joined game slot {}.", slot_name);
+                                            }
+                                            &RequestAction::LeaveGameSlot => {
+                                                client_state.in_game = false;
+                                                println!("Left game slot {}.", client_state.game_slot.unwrap());
+                                                client_state.game_slot = None;
+                                            }
+                                            _ => {
+                                                //XXX more cases in which server replies OK
+                                                println!("OK :)");
+                                            }
+                                        }
+                                    } else {
+                                        println!("OK, but we didn't make a request :/");
```

Comment:
Since this does sound impossible, should we make it `unreachable!()`?

Reply:
It isn't impossible. You can't trust the network. In any case, we'll be removing this when we add support for dropped/reordered/duplicated packets.